packages/relay/src/lib/eth.ts

```diff
@@ -1168,18 +1168,26 @@ export class EthImpl implements Eth {
       const contractCallResponse = await this.mirrorNodeClient.postContractCall(callData, requestId);
       return contractCallResponse && contractCallResponse.result ? EthImpl.prepend0x(contractCallResponse.result) : EthImpl.emptyHex;
     } catch (e: any) {
-      // Temporary workaround until mirror node web3 module implements the support of precompiles
-      // If mirror node throws, rerun eth_call and force it to go through the Consensus network
-      if (e) {
-        const errorTypeMessage = e instanceof MirrorNodeClientError && (e.isNotSupported() || e.isNotSupportedSystemContractOperaton()) ? 'Unsupported' : 'Unhandled';
-        this.logger.trace(`${requestIdPrefix} ${errorTypeMessage} mirror node eth_call request, retrying with consensus node`);
-
-        return await this.callConsensusNode(call, gas, requestId);
-      } 
-      this.logger.error(e, `${requestIdPrefix} Failed to successfully submit eth_call`);
       if (e instanceof JsonRpcError) {
         return e;
       }
+      
+      if (e instanceof MirrorNodeClientError) {
+        if (e.isRateLimit()) {
+          return predefined.IP_RATE_LIMIT_EXCEEDED(e.errorMessage || "eth_call");
+        }
+
```

Comment:
are we missing the `400` error check here?

Reply:
I think it's working because that check is already in `MirrorNodeClient.handleError()`
Might be cleaner to have it here but maybe for 0.24 to minimize moving things around.
However, if it's causing a gap or bug then yeah we can move or add a check here.
Adding a check would be harmless and address the concern

Reply:
Actually @AlfredoG87 i think you're right. We need to expose the contract reverted.
So there should be this check below

```ts
        if (e.isContractReverted()) {
          return predefined.CONTRACT_REVERT(e.errorMessage);
        }
```

Reply:
✅  Done,  also  added  UT to validate is working  correctly